### DISCUSSION TITLE
Failure when calling a method

### DISCUSSION THREAD

**Discussion by WiseLife42** [0x6d69636b/windows_hardening#143] [General]
Hello,

I tested the script on a Windows 10 virtual machine,
Everything went well when I used Audit, Hailmary and Config modes for my backup,
However, after using Hailmary, I told myself to test to go back to my basic configuration located in my "Backup.csv" file,
I get these error messages:

![image](https://user-images.githubusercontent.com/80531900/217553158-f968b093-45d9-40cf-a52e-a3e294f10144.png)

Here are the result statistics:
* Before HailMary mode: **Total checks: 349 - Passed: 71, Low: 58, Medium: 219, High: 1**
* After HailMary mode: **Total checks: 349 - Passed: 341, Low: 0, Medium: 7, High: 1**
* After re-importing my conf file "backup.csv": **Total checks: 349 - Passed: 107, Low: 34, Medium: 27, High: 1**

The commands i used : 
```
cd windows_hardening-master/
Set-ExecutionPolicy RemoteSigned
Import-Module .\HardeningKitty.psm1
# Before HailMary
Invoke-HardeningKitty -Mode Audit -Log 
Invoke-HardeningKitty -Mode Config -Backup -BackupFile Backup.csv
Invoke-HardeningKitty -Mode HailMary -Log -LogFile hailmary.txt -SkipRestorePoint -Report
# After HailMary 
Invoke-HardeningKitty -Mode Audit -Log 
Invoke-HardeningKitty -Mode HailMary -Log -LogFile hailmary.txt -SkipRestorePoint -FileFindingList .\Backup.csv
# After import Backup.csv
Invoke-HardeningKitty -Mode Audit -Log 
```

I use the signed script: https://github.com/scipag/HardeningKitty/blob/master/HardeningKitty.psm1
I use the French language for my OS

Unfortunately I don't have the same result anymore,
Thank you for clarifying me on the fact

**Comment by 0x6d69636b** [comment]
I have added a disclaimer in the README about the backup. At the moment, it is not possible to restore the 1:1 state with HardeningKitty alone. Registry settings can be restored easily, but there are no functions for other settings.

The ASR errors seem familiar to me. Can you test it with the development version? I thought I had fixed this.

**Comment by WiseLife42** [comment]
Thank you for the answer,

I have used the development script which can be found here: https://github.com/scipag/HardeningKitty/blob/master/HardeningKitty.psm1

Do you know when the 1:1 state will be available?

**Reply by 0x6d69636b** [reply]
Please use the version in this repo :) https://github.com/0x6d69636b/windows_hardening/blob/master/HardeningKitty.psm1

I have no idea when or even if this will be available. The effort to have a stable and reliable backup/restore function on Windows systems from Windows 10, 11 to Server 2012 until 2022 on different languages is enormous. Besides, there are numerous backup solutions, Windows Backup is integrated in the OS. I prefer to concentrate on new hardening settings and integrating new lists rather than putting all my effort into a fully developed backup solution. I recommend working with a test system (VM, container, image) and only when all settings and their implications have been understood, use this on productive systems. With more than 3 systems, I would personally rely on group policies and not use HailMary.

However, HardeningKitty is open source and if anyone wants to contribute a stable and fully functional backup/restore function, pull requests are welcome here ;-)
